ha/util/consul_kv_store.py

```diff
@@ -189,5 +270,80 @@ def get_keys(self, prefix):
             List of keys
         """
         if not ConsulKvStore._keys.get(prefix):
+
+            # get cached keys if batch put is enabled
+            payload_keys = None
+            if self._enable_batch_put:
+                # Using dictionary comprehension + startswith()
+                # Prefix key match in dictionary
+                payload_keys = [key for key in self._payload if key.startswith(prefix)]
+
             ConsulKvStore._keys[prefix] = self._consul.kv.get(prefix, keys=True)[1]
+
+            # Merge lists with unique values
+            if payload_keys:
+                ConsulKvStore._keys[prefix] = list(set(payload_keys + ConsulKvStore._keys[prefix]))
+
         return ConsulKvStore._keys[prefix]
+
+    def _kv_put_in_transaction(self, tx_payload: List[TxPutKV]):
+        """
+         This function put all the values from provided tuple list to consul server
+
+        Args:
+            tx_payload (List[TxPutKV]): list of Tuple that needs to put in consul server see definition 'TxPutKV'.
+        """
+        def to_payload(v: TxPutKV) -> Dict[str, Any]:
```

Comment:
What does function inside function definition do here ?

Reply:
I have coped as it is, it is getting called in this function only so ok to have inside only.